gzip_logs_tools/log_compressor/log_compressor.c

```diff
@@ -790,7 +796,9 @@ create_pid_file(const char *pid_file)
 	}	
 
 	sprintf(buf, "%ld\n", (long) getpid());
-	if (write(fd, buf, strlen(buf)) != strlen(buf))
+	size_t buf_len = strlen(buf);
+	ssize_t written_bytes = write(fd, buf, buf_len);
+	if (written_bytes < (ssize_t) buf_len)
```

Comment:
i'd prefer to keep it with `!=`, even though it can't be >, i prefer to be strict :)